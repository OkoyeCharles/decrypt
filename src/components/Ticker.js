import React from 'react';
import PropTypes from 'prop-types';
import '../styles/Ticker.css';
import { MdArrowForwardIos } from 'react-icons/md';
import { useDispatch } from 'react-redux';
import { useNavigate } from 'react-router';
import { currencyFormatter } from '../utils/index';
import { setDetails } from '../redux/actions';

function Ticker({
  id, symbol, price, change,
}) {
  const dispatch = useDispatch();
  const navigate = useNavigate();

  const showDetails = () => {
    dispatch(setDetails(id));
    navigate('/details');
  };

  return (
    <li className="ticker">
      <h3 className="ticker__symbol">{symbol}</h3>
      <div className="ticker__price">{currencyFormatter.format(price)}</div>
      <div
        className="ticker__change"
        style={parseFloat(change) > 0 ? { background: 'lightgreen' } : { background: '#ef340a98' }}
      >
        {`${change}%`}
      </div>
      <MdArrowForwardIos className="ticker__details" onClick={showDetails} />
    </li>
  );
}

Ticker.propTypes = {
  id: PropTypes.string.isRequired,
  symbol: PropTypes.string.isRequired,
  price: PropTypes.string.isRequired,
  change: PropTypes.string.isRequired,
};

export default Ticker;
