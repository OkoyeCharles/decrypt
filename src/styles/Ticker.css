.ticker {
  background: #fff;
  padding: 1em 0;
  display: grid;
  grid-template-columns: 1fr 1fr 1fr 0.5fr;
  place-items: center;
}

.ticker:nth-child(even) {
  background-color: #dcd5c1;
}

.ticker__symbol {
  color: #b58b03;
}

.ticker__change {
  padding: 0.25em 0.5em;
  border-radius: 0.25em;
}

.ticker__details {
  background: #efb90a;
  font-size: 1.5em;
  padding: 0.25em;
  border-radius: 50%;
}
