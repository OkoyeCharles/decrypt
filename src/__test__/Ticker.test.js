import { Provider } from "react-redux";
import { MemoryRouter } from "react-router";
import renderer from "react-test-renderer";
import Ticker from "../components/Ticker";
import store from "../redux/configureStore";

it("<Ticker /> renders correctly", () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <MemoryRouter>
            <Ticker id="80" symbol="BTC" price="20309.00" change="0.25" />
        </MemoryRouter>
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
