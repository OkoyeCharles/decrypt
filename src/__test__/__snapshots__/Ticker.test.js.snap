// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Ticker /> renders correctly 1`] = `
<li
  className="ticker"
>
  <h3
    className="ticker__symbol"
  >
    BTC
  </h3>
  <div
    className="ticker__price"
  >
    $20,309.00
  </div>
  <div
    className="ticker__change"
    style={
      Object {
        "background": "lightgreen",
      }
    }
  >
    0.25%
  </div>
  <svg
    className="ticker__details"
    fill="currentColor"
    height="1em"
    onClick={[Function]}
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 24 24"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M0 0h24v24H0V0z"
      fill="none"
    />
    <path
      d="M6.23 20.23L8 22l10-10L8 2 6.23 3.77 14.46 12z"
    />
  </svg>
</li>
`;
